### ISSUE TITLE
Syntax error gives RC=0 from compiler

### ISSUE THREAD

**Issue by kamahen** [SWI-Prolog/swipl-devel#826]
The compiler should return a non-zero return code if there's any error, including syntax errors. (For example, it has an option to return non-zero if there's an undefined predicate.)
```
$ swipl --stand_alone=true --undefined=error --verbose=false --foreign=save -o xxx.qlf -c xxx.pl ; echo rc=$?
ERROR: /tmp/xxx.pl:1:5: Syntax error: Operator expected
% Disabled autoloading (loaded 28 files)
% Disabled autoloading (loaded 2 files)
% Disabled autoloading (loaded 0 files)
rc=0
```

and the source for xxx.pl is:
```
this has a syntax error.
```

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/automatic-tests-can-succeed-with-syntax-errors/3891/5

**Comment by kamahen** [comment]
There's an ugly work-around, mentioned in the discussion group, by adding this to the code:
```
user:message_hook(Term, Type, _):-
    \+ started,
    ( Type = error ; Type = warning ),
    message_to_string(Term, String),
    writeln(user_error, String),
    halt(1).
```

**Comment by kamahen** [comment]
Fixed with Commit a09658cce2efa40bc805b99f21e0b11aa45e8584
